sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/UtilsImpl.java

```diff
@@ -466,6 +466,14 @@ public static long getBlobSize(HttpHeader contentRangeHeader) {
         throw LOGGER.logExceptionAsError(new ServiceResponseException("Invalid content-range header in response -" + contentRangeHeader));
     }
 
+    public static long getContentLength(HttpHeader contentLengthHeader) {
+        if (contentLengthHeader != null && contentLengthHeader.getValue() != null) {
+            return Long.parseLong(contentLengthHeader.getValue());
+        }
+
+        throw LOGGER.logExceptionAsError(new ServiceResponseException("Content-Length header in missing in the response"));
```

Comment:
Are there responses where we've noticed service not returning content length? I am expecting that this would never happen but good to handle this case. Should we throw an exception (resulting in a bit of a behavior change) or should we fall back to the existing behavior of  getting the length from BinaryData (i.e. fallback to buffering)?

Reply:
Good point! 
We're not using transfer-encoding, so service [SHOULD send](https://www.rfc-editor.org/rfc/rfc9110.html#name-content-length) content-length. So it'd indicate a bug on the service side and we've been told that that container registry should not be trusted and we should fail fast if unexpected things happen. 
This behavior was never released in a stable version, so it's ok to break.

Reply:
Makes sense to throw an exception then.